docs/Adding-Software-to-Homebrew.md

```diff
@@ -8,46 +8,139 @@ Before you start, please check the open pull requests for [Homebrew/homebrew-cor
+You're now ready to contribute your formula to the [homebrew-core](https://github.com/Homebrew/homebrew-core) repository. If this is your first time making a submission, consult the [How to Open a Homebrew Pull Request](How-To-Open-a-Homebrew-Pull-Request.md#formulae-related-pull-request) guide for detailed instructions. Your pull request will be reviewed by the maintainers, who may suggest improvements or changes before your formula can be added to Homebrew.
 
-If you've made it this far, congratulations on submitting a Homebrew formula! We appreciate the hard work you put into this and you can take satisfaction in knowing that your work may benefit other Homebrew users as well.
+If you've made it this far, congratulations on successfully submitting a Homebrew formula! Your dedication and hard work are highly valued. Take satisfaction in knowing that your contribution provides a valuable addition that will benefit many Homebrew users.
 
 ## Casks
 
 **Note:** Before taking the time to craft a new cask:
 
-* make sure it can be accepted by checking the [Rejected Casks FAQ](Acceptable-Casks.md#rejected-casks), and
+* make sure it can be accepted by checking [Acceptable Casks](Acceptable-Casks.md).
 * check that the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
+* if you are just updating the version of an existing cask, see the streamlined method in ['submit a new version of an existing cask'](How-To-Open-a-Homebrew-Pull-Request.md#submit-a-new-version-of-an-existing-cask).
 
 ### Writing the cask
 
-Making a new cask is easy. Follow the directions in [How to Open a Homebrew Pull Request](How-To-Open-a-Homebrew-Pull-Request.md#cask-related-pull-request) to begin.
+Making a new cask is easy, and the [Cask Cookbook](Cask-Cookbook.md) is an essential resource to guide you through the process. The Cookbook provides detailed instructions, best practices, and solutions to common problems you may encounter when creating a cask. Familiarizing yourself with its contents before you start is advisable.
+
+1. Begin by researching existing casks in Homebrew that are similar to the software you plan to add. This will help you understand the typical structure and conventions used in casks. Start by tapping `homebrew/cask`: run `brew tap homebrew/cask` to clone the `homebrew/cask` tap to the path returned by `brew --repository homebrew/cask`.
+
+1. If you're starting from scratch, you can use the [`brew create --cask <download-url>` command](Manpage.md#create-options-url) to produce a basic version of your cask.
+
+  After executing the `create` command, `EDITOR` will open with a cask template at `$(brew --repository homebrew/cask)/Casks/<prefix>/<cask>.rb` (where `<prefix>` is the first letter of your cask's name and `<cask>` is the name of your cask). The template will appear as follows:
```

Comment:
No need to state the specific location as it's an implementation detail that `brew create` and `brew edit` both output and handle for you

Reply:
This was extracted from / possibly slightly reworded from a section of the older Cask docs in this file.

The main reason I figured I would leave it here is that it easily shows where I can find the cask, so I don't need to wonder how to get my shell to that directory (for git/etc), or go look up those details somewhere else.

Reply:
> This was extracted from / possibly slightly reworded from a section of the older Cask docs in this file.

Yeh, I see that in hindsight. Not your bad but still worth improving given the line being changed 👍🏻 

> The main reason I figured I would leave it here is that it easily shows where I can find the cask, so I don't need to wonder how to get my shell to that directory (for git/etc), or go look up those details somewhere else.

This probably predated `brew create` and `brew edit` outputting the relevant path on run.

Reply:
@MikeMcQuaid Fixed in 204705bf76964844f1758271b7eeaa51e08d27a1

I'll leave this comment thread open for visibility for now; but once you check it and are happy, leave a comment and then feel free to mark it resolved.